src/markdown-pages/build-apps/customize-visualization-with-New-Relic-components.mdx

```diff
@@ -0,0 +1,408 @@
+              <RadarChart
+                width={width}
+                height={height}
+                data={transformedData}
+              >
+                <PolarGrid />
+                <PolarAngleAxis dataKey="name" />
+                  <PolarRadiusAxis tickFormatter={this.formatTick} />
+                  <Radar
+                    dataKey="value"
+                    stroke={stroke || '#51C9B7'}
+                    fill={fill || '#51C9B7'}
+                    fillOpacity={0.6}
+                  />
+              </RadarChart>
+            </React.Fragment>
+          );
+        }}
+      </NrqlQuery>
+    )}
+  </AutoSizer>
+);
+```
+
+</Steps>
+
+## Connect component state with the new UI controls
+
+In the next steps, we are going to add a function to update state and connect that with the `SegmentedControl` component.
+
+<Steps>
+
+  <Step>
+
+Add a component method that updates state to your visualization class.
+
+```js
+  updateSelectedChart = (evt, value) => {
+    this.setState({ selectedChart: value })
+  }
+```
+
+  </Step>
+
+  <Step>
+
+Use the `updateSelectedChart` method in `SegmentedControl` onChange prop.
+
+```jsx
+<SegmentedControl onChange={this.updateSelectedChart}>
+```
+
+  </Step>
+
+</Steps>
+
+<Steps>
+
+## Add our new chart option 
+
+In the next couple of steps, we will add the <Link to="https://recharts.org/en-US/api/Treemap">Treemap chart from Recharts</Link> and get the correct values passed into it.
+
+  <Step>
+
+Add `Treemap` to the import from recharts. 
+
+```js
+import {
+  Radar,
+  RadarChart,
+  PolarGrid,
+  PolarAngleAxis,
+  PolarRadiusAxis,
+  Treemap
+} from 'recharts';
+```
+    
+  </Step>
+
+To learn more about `Treemap` chart, go to the <Link to="https://recharts.org/en-US/examples/SimpleTreemap">Recharts Treemap example</Link> or the <Link to="https://recharts.org/en-US/api/Treemap">Recharts Treemap API docs</Link>.
+
+  <Step>
+
+Under the `RadarChart` closing tag in render, add the `Treemap` component and the neccessary props.
+
+```jsx
+<Treemap
+  width={width}
+  height={height}
+  data={transformedData}
+  dataKey="value"
+  ratio={4 / 3}
+  stroke="#fff"
+  fill="#8884d8"
+/>
+```
+
+  </Step>
+
+Now <Link to="/build-apps/build-visualization#render-the-visualization-in-local-development">render the visualziation in local development</Link> to see the new Treemap chart below your Radar chart.
+
+</Steps>
+
+<Steps>
+
+## Use component state to switch between charts
+
+In the final step, we will use our `selectedChart` state value to determine which chart to show, the Radar chart or the Treemap chart.
+
+  <Step>
+
+Destructure `selectedChart` from state near the top of the `render()` function.
+
+```js
+const { selectedChart } = this.state;
+```
+
+  </Step>
+
+  <Step>
+
+Wrap the `RadarChart` and `Treemap` in a ternary using the `selectedChart` state.
+
+```jsx
+{
+  selectedChart === CHART_TYPES.Radar ?
+  <RadarChart> ... <RadarChart>
+  :
+  <Treemap />
+}
+```
```

Comment:
It'd be helpful to do a little bit of hand-holding here and explain what the conditional does, in addition to the directive "wrap these things in a ternary":

> Here, you compare `selectedChart` to `CHART_TYPES.Radar`. If `selectedChart` is a `Radar`, you render a `RadarChart`. Otherwise, you render a `Treemap`.

Reply:
✅